scripts/lint_naming_conventions.py

```diff
@@ -0,0 +1,93 @@
+# not following conventions
+# https://github.com/rcbops/rpc-gating#naming-conventions
+
+import argparse
+import os
+import re
+import sys
+import yaml
+
+
+def parse_args():
+    description = "Parses JJB files to ensure naming conventions followed"
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument("--exclude-files",
+                        required=False,
+                        help="Comma seperated list of files to exclude")
+    parser.add_argument("--dirs",
+                        required=True,
+                        help="Comma seperated dirs to parse")
+    parser.add_argument("--parse-jjb", required=False,
+                        help="Parse jjb file for names",
+                        action='store_true')
+    return parser.parse_args()
+
+
+def parse_jjb_file(in_dir, in_file):
+    _rc = 0
+    filename = os.path.join(in_dir, in_file)
+    file_yaml = yaml.safe_load(open(filename).read())
+    for item in file_yaml:
+        if "project" in item:
+            # check project name
+            if parse_job_name(item["project"]["name"], filename):
+                _rc = 1
+            # check project jobs names
+            for job in item["project"]["jobs"]:
+                if parse_job_name(job, filename):
+                    _rc = 1
+
+        if "job-template" in item:
+            # check job-template name
+            if parse_job_name(item["job-template"]["name"], filename):
+                _rc = 1
+
+        if "job" in item:
+            # check individual job
+            if parse_job_name(item["job"]["name"], filename):
+                _rc = 1
+    return _rc
+
+
+def parse_job_name(job_name, file_name):
+    regex = '^([a-zA-Z0-9]+-?)+([_]([{][a-zA-Z0-9]+[}]-?)+)?$'
+    match = re.match(regex, job_name)
+    if not match:
+        out = "Lint error: Job name \"%s\" in \"%s\"" % (job_name, file_name)
+        out += "\n\tdoes not conform to rpc-gating naming conventions.\n"
+        sys.stderr.write(out)
+        return 1
+    return 0
+
+
+def parse_file_name(in_dir, in_file):
+    regex = '^([a-z0-9]+_?)+(\.{1}[a-z]+)?$'
+    filename = os.path.join(in_dir, in_file)
+    match = re.match(regex, in_file)
+    if not match:
+        out = "Lint error: filename \"%s\"\n\t does not " % filename
+        out += "conform to rpc-gating conventions\n"
+        sys.stderr.write(out)
+        return 1
+    return 0
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    rc = 0
+    for _dir in args.dirs.split(','):
+        for root, dirs, files in os.walk(_dir):
+            for _file in files:
+                if args.exclude_files and _file in args.exclude_files:
+                    pass
+                else:
+                    if args.parse_jjb and _file.endswith(".yml"):
+                        if parse_jjb_file(_dir, _file):
+                            rc = 1
+                    else:
+                        if parse_file_name(_dir, _file):
```

Comment:
Agreed, but I wanted to keep the `--parse-jjb` flag to act as a unique run only for job names on yml files, and not do it all at once. This way the directories being linted and exclude files could be changed if needed if another directory needs to be added for jobs. I figure the little bit of flexibility that offers is worth the second run of the script. Additionally this separates the lint to clearly be either job names, or file names.

In its current use it doesn't need that, and I can't imagine there's a great likelihood of this being required in the near future, so I'd be fine changing it if ifs not worth that seperation